app/services/replication_transfer_updater.rb

```diff
@@ -0,0 +1,150 @@
+# Copyright (c) 2015 The Regents of the University of Michigan.
+# All Rights Reserved.
+# Licensed according to the terms of the Revised BSD License
+# See LICENSE.md for details.
+
+class ReplicationTransferUpdater
+
+  class << self
+    def update(record, params)
+      build_update(record, params).update
+    end
+
+    def build_update(record, params)
+      case
+      when CancelUpdate.matching_update?(record, params)
+        CancelUpdate.new(record, params)
+      when FixityValueUpdate.matching_update?(record, params)
+        FixityValueUpdate.new(record, params)
+      when StoreRequestedUpdate.matching_update?(record, params)
+        StoreRequestedUpdate.new(record, params)
+      when StoredUpdate.matching_update?(record, params)
+        StoredUpdate.new(record, params)
+      else
+        ReplicationTransferUpdate.new(record, params)
+      end
+    end
+
+  end
+
+
+  # A generic update
+  class ReplicationTransferUpdate
+
+    # @param [ReplicationTransfer] record
+    # @param [HashWithIndifferentAccess] params The params hash of the update
+    def initialize(record, params)
+      @record = record
+      @params = params
+    end
+    def update
+      record.update(params)
+    end
+    def we_requested?
+      record.from_node&.namespace == Rails.configuration.local_namespace
+    end
+    def we_replicating?
+      record.to_node&.namespace == Rails.configuration.local_namespace
+    end
+
+    private
+    attr_reader :record, :params
+  end
```

Comment:
Classes using Node shouldn't have to know if this is the case or not.  I'd waffled between comparing the namespaces as is in the code now, and comparing the node identity, i.e.

```ruby
record.to_node == Node.local_node!
```
Where ::local_node is:
```ruby
  def self.local_node!
    self.find_by_namespace!(Rails.configuration.local_namespace)
  end
```

One of the issues with this method is it is always going to make a database call.  Perhaps an instance method `Node#is_local_node?` would be a better idiom.  Then any downcasing logic would go there, where it belongs.  Thoughts?